pkg/internal/data/ember.go

```diff
@@ -1,60 +1,17 @@
-package ember
+package data
 
 import (
 	"bytes"
-	"context"
 	_ "embed"
 	"encoding/csv"
-	"fmt"
 	"strconv"
-	"strings"
-
-	gridintensity "github.com/thegreenwebfoundation/grid-intensity-go/api"
 )
 
 //go:embed co2-intensities-ember-2021.csv
 var emberData []byte
 
-const ProviderName = "ember-climate.org"
-
-type ApiClient struct {
-	data map[string]GridIntensity
-}
-
-func New() (gridintensity.Provider, error) {
-	emberData, err := getGridIntensityData()
-	if err != nil {
-		return nil, err
-	}
-
-	return &ApiClient{data: emberData}, nil
-}
-
-func (a *ApiClient) GetCarbonIntensity(ctx context.Context, region string) (float64, error) {
-	result, ok := a.data[strings.ToUpper(region)]
-	if !ok {
-		return 0, fmt.Errorf("region %q not found", region)
-	}
-
-	return result.EmissionsIntensityGCO2PerKWH, nil
-}
-
-func GetGridIntensityForCountry(countryCode string) (*GridIntensity, error) {
-	emberData, err := getGridIntensityData()
-	if err != nil {
-		return nil, err
-	}
-
-	result, ok := emberData[strings.ToUpper(countryCode)]
-	if !ok {
-		return nil, fmt.Errorf("country code %q not found", countryCode)
-	}
-
-	return &result, nil
-}
-
-func getGridIntensityData() (map[string]GridIntensity, error) {
-	data := map[string]GridIntensity{}
+func GetEmberGridIntensity() (map[string]EmberGridIntensity, error) {
```

Comment:
This new `pkg/internal/data` package means the data and logic for getting the data file is separate from the provider.

We could extend this for other data files or to add more Ember data. The internal name is so its clear this isn't part of the public API.

Reply:
I hadn't thought of this as an idea, but I really like it. It's also something we could use on any other libraries that use local data. 👍